### PR TITLE
Add a full self deployment docker file

### PR DESCRIPTION
**Pull request by Pokeylooted**
Added a docker compose file that hosted all of movie-web, just to encourage more selfhosting and have a more straightforward way defined for people selfhosting.

### CODE REVIEW

**Comment by lonelil** [comment]
you don't need to add the backend url for cors for the backend, only the frontend.

**Comment by Pokeylooted** [comment]
> you don't need to add the backend url for cors for the backend, only the frontend.

Speaking of cors, do you know the cors situation on the proxy? I've had mixed results hosting it on that sometimes I have cors problems sometimes I don't

File: content/5.extra/2.selfhost.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+### Guide to full Self-Deployment of Movie-Web with Docker Compose
```
Comment (binaryoverload) [inline content/5.extra/2.selfhost.md:1]:
```suggestion
### Guide to full Self-Deployment of movie-web with Docker Compose
```

Sorry to be pedantic :P

File: content/5.extra/2.selfhost.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+   services:
+     postgres:
+       image: postgres
+       restart: unless-stopped
+       environment:
+         POSTGRES_USER: movie_web_user
+         POSTGRES_DB: movie_web_backend
+         POSTGRES_PASSWORD: YourPasswordHere
+       ports:
+         - "5432:5432"
```
Comment (binaryoverload) [inline content/5.extra/2.selfhost.md:24]: I don't think we need to expose the ports here, the backend can access it regardless as they are on the same network

File: content/5.extra/2.selfhost.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+       networks:
+         - movie-web-network
+
+     movie-web:
```
Comment (binaryoverload) [inline content/5.extra/2.selfhost.md:28]:
```suggestion
     movie-web-backend:
```

**Review by binaryoverload — CHANGES_REQUESTED** [review]
Overall good, just a few comments :)

**Comment by Pokeylooted** [comment]
Should be good to go now

File: content/5.extra/2.selfhost.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+       environment:
+         MWB_SERVER__CORS: "https://movie-backend.example.tld https://movie.example.tld"
+         MWB_SERVER__PORT: 8080
+         MWB_POSTGRES__CONNECTION: postgresql://movie_web_user:YourPasswordHere@postgres:5432/movie_web_backend
+         MWB_CRYPTO__SESSION_SECRET: 32CHARACTERLONGSECRET
+         MWB_META__NAME: Server name
+         MWB_META__DESCRIPTION: Server Description
+         MWB_POSTGRES__MIGRATE_ON_BOOT: "true"
+         MWB_SERVER__TRUSTPROXY: "true"
+         MWB_SERVER__TRUSTCLOUDFLARE: "true"
```
Comment (binaryoverload) [inline content/5.extra/2.selfhost.md:38]:
```suggestion
         MWB_SERVER__TRUSTCLOUDFLARE: "true"
         # This is needed to resolve errors running migrations on some platforms - does not affect the application 
         MIKRO_ORM_MIGRATIONS_DISABLE_FOREIGN_KEYS: "true"
```

File: content/5.extra/2.selfhost.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+         # This is needed to resolve errors running migrations on some platforms - does not affect the application 
+         MIKRO_ORM_MIGRATIONS_DISABLE_FOREIGN_KEYS: "true"
```
Comment (binaryoverload) [inline content/5.extra/2.selfhost.md:40]:
```suggestion
         MIKRO_ORM_MIGRATIONS_DISABLE_FOREIGN_KEYS: "false"
```

File: content/5.extra/2.selfhost.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+           OPENSEARCH_ENABLED: "true"
+           GA_ID: "Google ID Here"
+       ports:
+         - "80:80"
+       networks:
+         - movie-web-network
+       restart: unless-stopped
+
+     movie-web-proxy:
+       image: ghcr.io/movie-web/simple-proxy:2.1.4
```
Comment (binaryoverload) [inline content/5.extra/2.selfhost.md:67]:
```suggestion
       image: ghcr.io/movie-web/simple-proxy:latest
```

**Review by binaryoverload — APPROVED** [review]

**Pull request merged**